draft-ietf-rats-corim.md

```diff
@@ -2070,82 +2042,148 @@ Logically, new ECT entries are appended to the existing ACS.
+
+### Element map comparison {#sec-compare-element-map}
+
+A Verifier shall compare each `element-map` within the condition ECT `element-list` against the ACS entry `element-list`.
+
+First, a Verifier SHALL locate the entries in the ACS entry `element-list` which have a matching `element-id` as the condition ECT `element-map`.
+Two `element-id` fields are the same if they are either both omitted, or both present with binary identical deterministic encodings.
+
+Before performing the binary comparison, a Verifier SHOULD convert both fields into a form which meets CBOR Core Deterministic Encoding Requirements {{-cbor}}.
+
+If any condition ECT entry `element-id` does not have a corresponding `element-id` in the ACS entry then the element map does not match.
+
+If any condition ECT entry has multiple corresponding `element-id`s then the element map does not match.
+
+Second, a Verifier SHALL compare the `element-claims` field within the condition ECT `element-list` and the corresponding field from the ACS entry (see {{sec-compare-mvm}}).
+
+### Measurement values map map Comparison {#sec-compare-mvm}
+
+A Verifier SHALL iterate over the codepoints which are present in the condition ECT element's `measurement-values-map`.
+Each of the codepoints present in the condition ECT `measurement-values-map` is compared against the same codepoint in the candidate entry `measurement-values-map`.
+
+If any codepoint present in the condition ECT `measurement-values-map` does not have a corresponding codepoint within the candidate entry `measurement-values-map` then Verifier SHALL remove that candidate entry from the candidate entries array.
+
+If any codepoint present in the condition ECT `measurement-values-map` does not match the same codepoint within the candidate entry `measurement-values-map` then Verifier SHALL remove that candidate entry from the candidate entries array.
+
+#### Comparison of a single measurement-values-map codepoint {#sec-match-one-codepoint}
+
+A Verifier SHALL compare each condition ECT `measurement-values-map` value against the corresponding ACS entry value using the appropriate algorithm.
+
+Non-negative codepoints represent standard data representations.
+The comparison algorithms for these are defined in this document (in the sections below) or in other standards.
+For some non-negative codepoints their behavior is modified by the CBOR tag at the start of the condition ECT `measurement-values-map` value.
+
+Negative codepoints represent profile defined data representations.
+A Verifier SHALL use the codepoint number, the profile associated with the condition ECT, and the tag value (if present) to select the comparison algorithm.
+
+If a Verifier is unable to determine the comparison algorithm which applies to a codepoint then it SHALL behave as though the candidate entry does not match the condition ECT.
```

Comment:
Is this the right thing to do if there is no defined comparison algorithm?
For most of the standard (non-negative) codepoints in the document today, a binary compare is what would be expected.
As that is a non-editorial change I propose making it in a different PR.

Reply:
A profile may define only codepoints -1 to -15. If there's an update and -16 creeps in without the profile also getting a version update, do we want to say that additions are non-breaking, or that all changes require version updates and therefore new profile ids? I'd like that to be up to the profile designer, honestly, which would be within its rights to say that all code points below -15 or between -15 and -1000 are compared with binary equality to allow for opaque expansion but still allowing for a minor version advancement to add non-opaque comparisons to new code points.

I'd say no match. Keep this text.

Reply:
I'm OK with the current text.